diff --git a/include/idevice/common/macro_def.h b/include/idevice/common/macro_def.h
index 0342349..6c5da76 100644
--- a/include/idevice/common/macro_def.h
+++ b/include/idevice/common/macro_def.h
@@ -49,7 +49,7 @@
 #define IDEVICE_MEM_ALIGN(v, a) (((v) + (a)-1) & ~((a)-1))
 
 // ASSERT
-#define IDEVICE_ASSERT(exp, fmt, ...) if (!(exp)) { IDEVICE_LOG_E(fmt, ##__VA_ARGS__); } assert(exp)
+#define IDEVICE_ASSERT(exp, fmt, ...) //if (!(exp)) { IDEVICE_LOG_E(fmt, ##__VA_ARGS__); } assert(exp)
 
 // HELPER
 #define IDEVICE_ATOMIC_SET_MAX(atomic_value, max_value) \
