diff --git a/include/usbmuxd.h b/include/usbmuxd.h
index 859fccd..bb8e88c 100644
--- a/include/usbmuxd.h
+++ b/include/usbmuxd.h
@@ -341,6 +341,14 @@ USBMUXD_API void libusbmuxd_set_debug_level(int level);
  */
 USBMUXD_API const char* libusbmuxd_version();
 
+/**
+ * Connect to remote device
+ * @param ip address of target device.
+ * @param port of target device.
+ */
+int usbmuxd_connect_remote(const char* ip_address, uint16_t port);
+void usbmuxd_disconnect_remote();
+
 #ifdef __cplusplus
 }
 #endif
diff --git a/src/libusbmuxd.c b/src/libusbmuxd.c
index 6b7d220..dfa74fd 100644
--- a/src/libusbmuxd.c
+++ b/src/libusbmuxd.c
@@ -142,6 +142,24 @@ static struct collection listeners;
 thread_once_t listener_init_once = THREAD_ONCE_INIT;
 mutex_t listener_mutex;
 
+static char* remote_ip = "";
+static uint16_t remote_port = 0;
+static unsigned char use_remote = 0;
+int usbmuxd_connect_remote(const char* ip_address, uint16_t port)
+{
+	remote_ip = strdup(ip_address);
+	remote_port = port;
+	int res = socket_connect(remote_ip, remote_port);
+	use_remote = res >= 0 ? 1 : 0;
+	return res;
+}
+void usbmuxd_disconnect_remote()
+{
+	free(remote_ip);
+	remote_port = 0;
+	use_remote = 0;
+}
+
 /**
  * Finds a device info record by its handle.
  * if the record is not found, NULL is returned.
@@ -223,6 +241,9 @@ static int connect_usbmuxd_socket()
 #else
 	res = socket_connect_unix(USBMUXD_SOCKET_FILE);
 #endif
+	if (use_remote > 0) {
+		res = socket_connect(remote_ip, remote_port);
+	}
 	if (res < 0) {
 		res = -errno;
 	}
